waitress/static/js/dashboard.js

```diff
@@ -0,0 +1,91 @@
+// DOM elements
+const modalBody = document.getElementById('modal-body');
+const refreshSlackUserBtn = document.getElementById('refresh-slack-btn');
+
+
+const notyConfig = {
+    timeout: 1000,
+    closeWith: ['click'],
+    theme: 'relax'
+};
+
+const SPINNER_HTML = `<section class="d-flex justify-content-center align-items-center w-100">
+<div class="spinner-grow text-dark"></div>
+</section>`;
+
+const REFRESH_HTML = '<i class="fas fa-sync-alt"></i>';
+
+export async function refreshSlackUser() {
+    try {
+        refreshSlackUserBtn.innerHTML = `${SPINNER_HTML} Loading...`;
+        refreshSlackUserBtn.disabled = true;
+        await axios.put('/users/update-users/');
+        sendNotification('Slack users Refreshed Successfully');
+        refreshSlackUserBtn.innerHTML = REFRESH_HTML;
+        refreshSlackUserBtn.disabled = false;
+    } catch(error) {
+        sendNotification('error', error.message);
+        console.log(error.message);
+        refreshSlackUserBtn.innerHTML = REFRESH_HTML;
+        refreshSlackUserBtn.disabled = false;
+    }
+}
+
+function sendNotification(text, type = 'success') {
+    return new Noty({ text, type, ...notyConfig }).show();
+}
+
+function generateModalHeader(title) {
+    return `<!-- Modal Header -->
+<div class="modal-header">
+    <h4 class="modal-title">${title}</h4>
+    <button type="button" class="close" data-dismiss="modal">&times;</button>
+</div>`;
+}
+
+const updateModalFooter = `<!-- Modal footer -->
+<div class="modal-footer">
+    <button type="button" class="btn btn-primary">Update</button>
+    <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
+</div>`;
+
+export function updateUserInfo(userId, firstname, lastname, is_active) {
+    const isActive = (is_active === 'True');
+    const modalHeader = generateModalHeader('Update User Info');
+    modalBody.innerHTML = `${modalHeader}
```

Comment:
User innerHtml is considered bad and a giant security flaw, why don't you create html instead or sanitze them as you see fit